### PR TITLE
add unit tests for customers

### PR DESCRIPTION
**Pull request by Raid5594**
- added mock for ddc-customers
- added unit tests for ddc-customers

### CODE REVIEW

**Review by yahortsaryk — DISMISSED** [review]

File: pallets/ddc-customers/src/mock.rs [hunk 1]
```diff
@@ -0,0 +1,154 @@
+//! Test utilities
+
+#![allow(dead_code)]
```
Comment (aie0) [inline pallets/ddc-customers/src/mock.rs:3]: don't use it globally but for specific line - even in mock/tests

File: pallets/ddc-customers/Cargo.toml [hunk 2]
```diff
@@ -11,10 +11,15 @@ frame-support = { version = "4.0.0-dev", default-features = false, git = "https:
 frame-system = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.30" }
 log = { version = "0.4.17", default-features = false }
 scale-info = { version = "2.1.1", default-features = false, features = ["derive"] }
+sp-io = { version = "6.0.0", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.30" }
```
Comment (aie0) [inline pallets/ddc-customers/Cargo.toml:14]: why do you need this? there is no changes to lib.rs so it must be related to mock/tests?

can this be moved to dev-dependencies?

File: pallets/ddc-customers/src/tests.rs [hunk 3]
```diff
@@ -0,0 +1,163 @@
+//! Tests for the module.
+
+use super::{mock::*, *};
+use ddc_primitives::ClusterId;
+
+use frame_support::{assert_noop, assert_ok};
+use pallet_balances::Error as BalancesError;
+
+#[test]
+fn create_bucket_works() {
```
Comment (aie0) [inline pallets/ddc-customers/src/tests.rs:10]: add another test to create 2 buckets and that 2 buckets can co-exist

File: pallets/ddc-customers/src/tests.rs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+	ExtBuilder.build_and_execute(|| {
+		System::set_block_number(1);
+
+		// Deposit dust
+		assert_noop!(
+			DdcCustomers::deposit(RuntimeOrigin::signed(1), 0_u128),
+			Error::<Test>::InsufficientDeposit
+		);
+
+		// Deposit all tokens fails (should not kill account)
```
Comment (aie0) [inline pallets/ddc-customers/src/tests.rs:43]: add usecase where account gets killed

**Review by aie0 — CHANGES_REQUESTED** [review]

File: pallets/ddc-customers/src/mock.rs (hunk 1 above)
Comment (Raid5594) [reply pallets/ddc-customers/src/mock.rs]: +

File: pallets/ddc-customers/Cargo.toml (hunk 2 above)
Comment (Raid5594) [reply pallets/ddc-customers/Cargo.toml]: done

File: pallets/ddc-customers/src/tests.rs (hunk 3 above)
Comment (Raid5594) [reply pallets/ddc-customers/src/tests.rs:10]: done

File: pallets/ddc-customers/src/tests.rs (hunk 4 above)
Comment (Raid5594) [reply pallets/ddc-customers/src/tests.rs:76]: discussed

**Review by aie0 — APPROVED** [review]

**Review by yahortsaryk — APPROVED** [review]

**Pull request merged**